exams/01. Ski Lift/app.js

```diff
@@ -0,0 +1,147 @@
+window.addEventListener('load', solve);
+
+function solve() {
+    const firstNameElement = document.querySelector('#first-name');
+    const lastNameElement = document.querySelector('#last-name');
+    const peopleCountElement = document.querySelector('#people-count');
+    const fromDateElement = document.querySelector('#from-date');
+    const daysElement = document.querySelector('#days-count');
+
+    const previewInfoUl = document.querySelector('ul.ticket-info-list');
+    const confirmInfoUl = document.querySelector('ul.confirm-ticket');
+
+    const button = document.querySelector('#next-btn');
+
+    button.addEventListener("click", previewTicket);
+
+    function previewTicket(e) {
+        e.preventDefault();
+
+        const firstName = firstNameElement.value;
+        const lastName = lastNameElement.value;
+        const peopleCount = peopleCountElement.value;
+        const fromDate = fromDateElement.value;
+        const days = daysElement.value;
+
+        if (!firstName || !lastName || !peopleCount || !fromDate || !days) {
+            return;
+        }
+
+        emptyInput(firstNameElement, lastNameElement, peopleCountElement, fromDateElement, daysElement);
+        createListItem(previewInfoUl, firstName, lastName, peopleCount, fromDate, days);
+    }
+
+    function emptyInput(firstNameElement, lastNameElement, peopleCountElement, fromDateElement, daysElement) {
+        firstNameElement.value = '';
+        lastNameElement.value = '';
+        peopleCountElement.value = '';
+        fromDateElement.value = '';
+        daysElement.value = '';
+        button.disabled = true;
+    }
+    
+    function createListItem(previewInfoUl, firstName, lastName, peopleCount, fromDate, days) {
+        const listElement = document.createElement('li');
+        listElement.classList.add('ticket');
+        previewInfoUl.appendChild(listElement);
```

Comment:
Why passing the previewInfoUl as parameter and not getting it here directly? If an element is not going to be passed across many functions I would reccomend to declare it inside the function which is going to use it.

Reply:
Just now i noticed i have declared the previewInfoUl in the closure and the pointless passing it to the function arguments. Thanks !